lib/Option/SanitizerOptions.cpp

```diff
@@ -108,33 +120,34 @@ llvm::SanitizerCoverageOptions swift::parseSanitizerCoverageArgValue(
   return opts;
 }
 
-static bool isTSanSupported(
-    const llvm::Triple &Triple,
-    llvm::function_ref<bool(llvm::StringRef)> sanitizerRuntimeLibExists) {
-
-  return Triple.isArch64Bit() && sanitizerRuntimeLibExists("tsan");
-}
-
 OptionSet<SanitizerKind> swift::parseSanitizerArgValues(
     const llvm::opt::ArgList &Args,
     const llvm::opt::Arg *A,
     const llvm::Triple &Triple,
     DiagnosticEngine &Diags,
-    llvm::function_ref<bool(llvm::StringRef)> sanitizerRuntimeLibExists) {
+    llvm::function_ref<bool(llvm::StringRef, bool)> sanitizerRuntimeLibExists) {
   OptionSet<SanitizerKind> sanitizerSet;
 
   // Find the sanitizer kind.
   for (int i = 0, n = A->getNumValues(); i != n; ++i) {
-    StringRef opt = A->getValue(i);
-    if (opt == "address") {
-      sanitizerSet |= SanitizerKind::Address;
-    } else if (opt == "thread") {
-      sanitizerSet |= SanitizerKind::Thread;
-    } else if (opt == "fuzzer") {
-      sanitizerSet |= SanitizerKind::Fuzzer;
-    } else {
+    auto kind = llvm::StringSwitch<Optional<SanitizerKind>>(A->getValue(i))
+        .Case("address", SanitizerKind::Address)
+        .Case("thread", SanitizerKind::Thread)
+        .Case("fuzzer", SanitizerKind::Fuzzer)
+        .Default(None);
+    bool isShared = kind && *kind != SanitizerKind::Fuzzer;
+    if (!kind) {
       Diags.diagnose(SourceLoc(), diag::error_unsupported_option_argument,
           A->getOption().getPrefixedName(), A->getValue(i));
+    } else if (sanitizerRuntimeLibExists(toFileName(*kind), isShared)
+        && (*kind != SanitizerKind::Thread || Triple.isArch64Bit())) {
```

Comment:
Can we keep the helper function "isTSanSupported"? Even if it's just a test for .isArch64bit. This condition is very hard to read.

Reply:
@kubamracek keeping the function does not make sense, as we now check existence of all sanitizers. I've tried to rewrite the condition to make it more readable.

Reply:
Can we then just be a bit more verbose here? Something like
```
bool sanitizerSupported = true;
if (*kind == Thread) {
  if (!Triple.is64bit() sanitizerSupported = false; // TSan requires 64-bit architecture
}
...
```